pandas/core/indexes/base.py

```diff
@@ -571,11 +580,11 @@ def _simple_new(cls, values, name: Hashable = None):
         return result
 
     @cache_readonly
-    def _constructor(self):
+    def _constructor(self: _IndexT) -> Type[_IndexT]:
```

Comment:
does it matter that a couple of subclasses behave differently?

Reply:
Ive typed the subclass methods, I dont think this is a problem.